workflow/controller/operator.go

```diff
@@ -1704,7 +1706,7 @@ func (woc *wfOperationCtx) deletePVCs(ctx context.Context) error {
 
 	switch gcStrategy {
 	case wfv1.VolumeClaimGCOnSuccess:
-		if woc.wf.Status.Phase == wfv1.WorkflowError || woc.wf.Status.Phase == wfv1.WorkflowFailed {
+		if woc.wf.Status.Phase == wfv1.WorkflowError || woc.wf.Status.Phase == wfv1.WorkflowFailed || woc.wf.Status.Phase == wfv1.WorkflowRunning {
```

Comment:
this makes me wonder why the original logic wasn't just simply `woc.wf.Status.Phase != wfv1.WorkflowSucceeded`

Reply:
I guess the original function was written with the assumption that it would only ever be called if the Workflow were in fact Completed. If we want to make it safer to call this even when the Workflow isn't completed, then we should also add a check to the `wfv1.VolumeClaimGCOnCompletion` case.